lib/core/include/irods/authenticate.h

```diff
@@ -0,0 +1,61 @@
+#ifndef IRODS_AUTHENTICATE_H
+#define IRODS_AUTHENTICATE_H
+
+/// \file
+
+struct RcComm;
+
+// These macros are used extensively throughout the authentication code. Over time, we should move away from using
+// these, but we cannot escape from it for now. Just know that these values are currently very important.
+// NOLINTBEGIN(cppcoreguidelines-macro-usage)
+#define MAX_PASSWORD_LEN 50
+#define CHALLENGE_LEN    64
+#define RESPONSE_LEN     16
+// NOLINTEND(cppcoreguidelines-macro-usage)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// \brief Authenticate a client with a connected iRODS server.
+///
+/// After returning successfully, the client will be authenticated with the connected iRODS server for API requests.
+///
+/// \param[in] _comm The iRODS communication object.
+/// \param[in] _context A string representing a JSON object. \parblock
+///
+/// The JSON object is used in the authentication plugin framework. The inputs will differ based on the authentication
+/// scheme, and not all options are recognized by all authentication schemes. The authentication scheme can be provided
+/// with the "scheme" key and a value of a string containing the name of the scheme to use (e.g. "native"). If no such
+/// key is provided, the authentication scheme will be derived from the local client environment.
+///
+/// The JSON object must specify the authentication scheme to use. Minimally, it should look like this:
+/// \code{.js}
+/// {"scheme": "<authentication scheme name>"}
+/// \endcode
+///
+/// If the authentication scheme is not provided in \p _context, an error will be returned.
```

Comment:
This seems to be in conflict with L29-L30 too?

Reply:
Fixed L29-L30. That was left over.